File: project-euler/problem-6/solution-6.js
function getSumOfSquares(n) {
  return (n * (n + 1) * (2 * n + 1)) / 6;
}

function getSquareOfSum(n) {
  return Math.pow((n * (n + 1)) / 2, 2);
}

function getDifference(n) {
  return Math.abs(getSumOfSquares(n) - getSquareOfSum(n));
}

console.log(getDifference(10)); // 2640
console.log(getDifference(100)); // 25164150

File: project-euler/problem-6/solution-6.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

long getSquareOfSum(int n)
{
    return pow((n * (n + 1)) / 2, 2);
}

long getSumOfSquare(int n)
{
    return (n * (n + 1) * ((2 * n) + 1)) / 6;
}

int main()
{
    register int n = 100;
    printf("%d", abs(getSumOfSquare(n) - getSquareOfSum(n))); // 25164150
    return 0;
}
